<launch>
  <!-- Override if you are not using the bottom USB connector but an internal serial port. -->
  <arg name="serial_port" default="/dev/ttyACM0" doc="Serial port to use for communicating with robot platform." />

  <!-- urdf xml robot description loaded on the Parameter Server-->
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find am_description)/urdf/automower.urdf.xacro'" />

  

  <!-- Start the am_driver -->
  <node name="am_driver_safe" pkg="am_driver_safe" type="am_driver_safe_node" output="screen">
    <param name="serialPort" value="$(arg serial_port)" type="str" />
    <param name="printCharge" value="false"/>

    <param name="updateRate" value="50.0"/>

    <param name="GPSCheckFreq" value="1"/>
    <param name="sensorStatusCheckFreq" value="5"/>
    <param name="encoderSensorFreq" value="10"/>
    <param name="batteryCheckFreq" value="1"/>
    <param name="loopSensorFreq" value="1"/>
    <param name="wheelSensorFreq" value="50"/>
    <param name="regulatorFreq" value="50"/>
    <param name="setPowerFreq" value="0"/>
    <param name="pitchRollFreq" value="10"/>
    <param name="stateCheckFreq" value="1"/>

    <param name="publishTf" value="1"/>
    <param name="velocityRegulator" value="1"/>
    <param name="pitchAndRoll" value="true"/>
    <param name="publishEuler" value="true"/>
    <param name="startWithoutLoop" value="true"/>

    <param name="jsonFile" value="$(find am_driver_safe)/config/automower_hrp.json" type="string" />
  </node>


  <!-- source that publishes the joint positions as a sensor_msgs/JointState -->
  <param name="use_gui" value="false"/>
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

  <!-- publish all the frames to TF -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
    <param name="publish_frequency" value="20"/> <!-- Hz -->
  </node>

</launch>
